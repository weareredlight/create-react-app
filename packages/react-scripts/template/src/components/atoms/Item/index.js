import React from 'react';
import PropTypes from 'prop-types';

import styles from './style.scss';

const Item = ({ onClick, elem }) => {
  const onRemove = () => onClick(elem.id);
  return (
    <li className={styles.item}>
      {elem.id} ~ {elem.name}
      <button onClick={onRemove}> X </button>;
    </li>
  );
};

Item.propTypes = {
  elem: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Item;
