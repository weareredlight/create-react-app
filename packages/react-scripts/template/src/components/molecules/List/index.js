import React from 'react';
import PropTypes from 'prop-types';

import Item from './../../atoms/Item/index';

const List = props => (
  <ul>
    {props.names.map(elem => (
      <Item key={elem.id} onClick={props.onRemove} elem={elem} />
    ))}
  </ul>
);

List.propTypes = {
  names: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
  ),
  onRemove: PropTypes.func.isRequired,
};

export default List;
