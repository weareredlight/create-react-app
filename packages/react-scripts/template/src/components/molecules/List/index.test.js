import React from 'react';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import { expect as expectChai } from 'chai';
import Adapter from 'enzyme-adapter-react-16';

import List from './index';
import Item from './../../atoms/Item/index';

Enzyme.configure({ adapter: new Adapter() });

const names = [
  { id: 1, name: 'Andre' },
  { id: 2, name: 'Joao' },
  { id: 3, name: 'Rui' },
  { id: 4, name: 'Sara' },
];

it('Renders all children', () => {
  const wrapper = shallow(<List names={names} onRemove={() => null} />);

  expectChai(wrapper.find(Item).length).to.equal(names.length);
  expect(toJson(wrapper)).toMatchSnapshot();
});
