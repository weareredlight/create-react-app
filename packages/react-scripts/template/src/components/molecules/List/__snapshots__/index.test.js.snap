// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders all children 1`] = `
<ul>
  <Item
    elem={
      Object {
        "id": 1,
        "name": "Andre",
      }
    }
    key="1"
    onClick={[Function]}
  />
  <Item
    elem={
      Object {
        "id": 2,
        "name": "Joao",
      }
    }
    key="2"
    onClick={[Function]}
  />
  <Item
    elem={
      Object {
        "id": 3,
        "name": "Rui",
      }
    }
    key="3"
    onClick={[Function]}
  />
  <Item
    elem={
      Object {
        "id": 4,
        "name": "Sara",
      }
    }
    key="4"
    onClick={[Function]}
  />
</ul>
`;
